internal/impl/kafka/franz_reader_ordered.go

```diff
@@ -389,13 +421,50 @@ func (f *FranzReaderOrdered) Connect(ctx context.Context) error {
 		)
 	}
 
-	if cl, err = kgo.NewClient(clientOpts...); err != nil {
+	if f.client, err = kgo.NewClient(clientOpts...); err != nil {
 		return err
 	}
 
+	// Check connectivity to cluster
+	if err = f.client.Ping(ctx); err != nil {
+		return fmt.Errorf("failed to connect to cluster: %s", err)
+	}
+
+	if f.lagUpdater != nil {
+		f.lagUpdater.Stop()
+	}
+	adminClient := kadm.NewClient(f.client)
+	f.lagUpdater = asyncroutine.NewPeriodic(f.topicLagRefreshPeriod, func() {
```

Comment:
I think you want `asyncroutine.NewPeriodicWithContext`

Reply:
Nice catch!